### PR TITLE
Gitlab 332: GetMailboxes should list delegated mailboxes

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java [hunk 1]
```diff
@@ -554,4 +554,106 @@ public void getMailboxesShouldReturnMailboxesWithFullSharedWithWhenHasAdminRight
...
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(ARGUMENTS + ".list", hasSize(6))
+            .body(ARGUMENTS + ".list.name", hasItems(expectedMailboxes.toArray()));
+    }
+
+    @Test
+    public void getMailboxesShouldReturnAllPrivateMailboxesAndAllSharedMailboxesWithReadOnlyWhenEmptyIds() throws Exception {
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: getMailboxesShouldFilterMailboxesWithoutReadRight 

I believe

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java [hunk 2]
```diff
@@ -554,4 +554,106 @@ public void getMailboxesShouldReturnMailboxesWithFullSharedWithWhenHasAdminRight
 
         assertThat(sharedWith).containsOnlyKeys(alice, cedric);
     }
+
+    @Test
+    public void getMailboxesShouldReturnAllPrivateMailboxesAndAllSharedMailboxesWhenEmptyIds() throws Exception {
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: getMailboxesShouldReturnAllAccessibleMailboxesWhenEmptyIds

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java [hunk 3]
```diff
@@ -554,4 +554,106 @@ public void getMailboxesShouldReturnMailboxesWithFullSharedWithWhenHasAdminRight
...
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(ARGUMENTS + ".list", hasSize(6))
+            .body(ARGUMENTS + ".list.name", hasItems(expectedMailboxes.toArray()));
+    }
+
+    @Test
+    public void getMailboxesShouldReturnExactUserInbox() throws Exception {
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: IMO you can remove this test, it seems already covered....

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java [hunk 4]
```diff
@@ -554,4 +554,106 @@ public void getMailboxesShouldReturnMailboxesWithFullSharedWithWhenHasAdminRight
...
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(ARGUMENTS + ".list.name", hasItems(MailboxConstants.INBOX))
+            .body(ARGUMENTS + ".list.id", hasItems(aliceInboxMailbox.serialize()));
+    }
+
+    @Test
+    public void getMailboxesShouldReturnSharedMailboxesWithRead() throws Exception {
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: Is this already covered?

If so, we can remove it...

**Comment by chibenwa** [comment]
BTW rebase it on MAILBOX-310

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java (hunk 3 above)
Comment (rouazana) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: I don't think it is covered, as here you are adding a mailbox share with the same name.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java (hunk 4 above)
Comment (rouazana) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: it's adding filtering to the previous test, so again I think it's not already covered

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java (hunk 3 above)
Comment (quynhn) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: We have two different INBOX mailbox in this case then we test result contain right user's INBOX
This case is described on ticket so I would like to keep this testing

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java (hunk 4 above)
Comment (quynhn) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java]: This one for my debugging but I would like to keep it such as it makes sence, user can access to specified share mailbox.

**Comment by chibenwa** [comment]
test this please

**Comment by strokyl** [comment]
Aren't we supposed to never use internal ticket number on Github but apache's jira number ?

**Comment by rouazana** [comment]
> Aren't we supposed to never use internal ticket number on Github but apache's jira number ?

We must not used gitlab numbers in commit.
We should not use gitlab numbers in PR, but that's not a big concern as our gitlab is open

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**